io/disk/softwareraid.py

```diff
@@ -40,116 +39,58 @@ class SoftwareRaid(Test):
-                self.extensivetest()
-
-    def create_raid(self):
-        """
-        Only basic operations are run viz create and delete
-        """
-        cmd = "echo 'yes' | mdadm --create --verbose --assume-clean \
-            %s --level=%s --raid-devices=%d %s --metadata %s" \
-            % (self.raid, self.raidlevel, self.disk_count, self.disk,
-               self.metadata)
-        if self.sparedisk:
-            cmd += " --spare-devices=1 %s " % self.sparedisk
-        if self.force:
-            cmd += " --force"
-        self.check_pass(cmd, "Failed to create a MD device")
-        self.mdadm_detail()
-
-    def extensivetest(self):
-        """
-        Extensive software raid options are run viz create, delete, assemble,
-        create spares, remove and add drives
-        """
-        cmd = "mdadm --fail %s %s" % (self.raid, self.remadd)
-        self.check_pass(cmd, "Unable to fail a drive from MD device")
-        self.mdadm_detail()
-        cmd = "mdadm --manage %s --remove %s" % (self.raid, self.remadd)
-        self.check_pass(cmd, "Failed to remove a drive from MD device")
-        self.mdadm_detail()
-        cmd = "mdadm --manage %s --add %s" % (self.raid, self.remadd)
-        self.check_pass(cmd, "Failed to add back the drive to MD device")
-        self.mdadm_detail()
-        cmd = "mdadm --manage %s --stop" % self.raid
-        self.check_pass(cmd, "Failed to stop/remove the MD device")
-        cmd = "mdadm --assemble %s %s %s" \
-              % (self.raid, self.disk, self.sparedisk)
-        self.check_pass(cmd, "Failed to assemble back the MD device")
-        while self.is_mdadm_recovering():
-            time.sleep(30)
-        process.system(cmd, ignore_status=True, shell=True)
-        self.mdadm_detail()
-
-    def mdadm_detail(self):
-        """
-        Function to print the details of mdadm array
-        """
-        cmd = "mdadm --detail %s" % self.raid
-        output = process.run(cmd, ignore_status=True, shell=True)
-        if output.exit_status != 0:
-            self.fail("Failed to display MD device details")
-        return output.stdout.decode("utf-8").splitlines()
-
-    def is_mdadm_recovering(self):
-        """
-        Function to check if the array is recovering or not
-        """
-        for line in self.mdadm_detail():
-            if 'State' in line and 'recovering' in line:
-                return True
-        return False
-
-    def check_pass(self, cmd, errmsg):
-        """
-        Function to check if the cmd is successful or not, if not display
-        appropriate message
-        """
-        if process.system(cmd, ignore_status=True, shell=True) != 0:
-            self.fail(errmsg)
+        disk = self.params.get('disks', default='').strip(" ")
+        if not disk:
+            self.cancel('No disks given')
+        required_disks = self.params.get('required_disks', default=1)
+        disk = disk.split()
```

Comment:
everything looks good for me other than few minor comments on variable namings.

disk --->> disks